### PR TITLE
Staging clx todd 081221

### PR DESCRIPTION
**Pull request by toddcd**
# JIRA

[CLXD-1396](https://clxmedia.atlassian.net/browse/CLXD-1396)
[CLXD-1467](https://clxmedia.atlassian.net/browse/CLXD-1467)

Related but not contained here is the following data migration script
[CLXD-1533](https://clxmedia.atlassian.net/browse/CLXD-1533)

# Description
Merging the above into QA for integration testing. Initial testing and user acceptance was done on my dev branch

### CODE REVIEW

File: clx_crm/models/crm_lead.py [hunk 1]
```diff
@@ -1,16 +1,227 @@
...
+                    else:
+                        results = contact_table.search(
+                            ["&", ("name", "ilike", search_name), ("company_type", "=", "person")]
+                        )
+                        if not results:
+                            contact.update({"existing_contact_id": 0, "validated": True})
+
+                        for existing in results:
+                            lead = self._build_lead_contact_validation(contact, existing)
+                            contact_list.append(lead)
```
Comment (dmontooth) [inline clx_crm/models/crm_lead.py:144]: The if not results: through the existing results blocks could be moved out of the if else blocks and run after results is set. Reduces a bit of code.

File: clx_crm/static/src/js/clx_crm.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                let exportButton = existingContactTree.closest('.modal-content').find('.o_list_export_xlsx');
+
+                // add handlers
+                selectedContactsBtn && selectedContactsBtn.click(this.proxy('saveSelectedContactOptions'));
+
+                // Remove unused search bar elements from header
+                exportButton.hide();
+                existingContactTree.closest('.modal-content').find('.o_cp_left').hide();
+                existingContactTree.closest('.modal-content').find('.o_cp_right').hide();
+                existingContactTree.closest('.modal-content').find('.o_searchview').hide();
```
Comment (dmontooth) [inline clx_crm/static/src/js/clx_crm.js:46]: Minor optimization - You can find and hide multiple using a comma-separated list

File: clx_ratio_invoice/static/src/js/clx_ratio_invoice.js [hunk 3]
```diff
@@ -35,9 +42,9 @@ odoo.define('clx_ratio_invoice.clx.ratio.invoice.template', function (require) {
                 });
 
                 // Remove unused search bar elements from header
-                $('.o_cp_left').hide();
-                $('.o_cp_right').hide();
-                $('.o_searchview').hide();
+                coopTree.closest('.modal-content').find('.o_cp_left').hide();
+                coopTree.closest('.modal-content').find('.o_cp_right').hide();
+                coopTree.closest('.modal-content').find('.o_searchview').hide();
```
Comment (dmontooth) [inline clx_ratio_invoice/static/src/js/clx_ratio_invoice.js:47]: Same minor optimization note. Find and hide multiple at once.

**Review by dmontooth — APPROVED** [review]
I added a couple of minor things that aren't blockers but could be optimized.

File: clx_crm/models/crm_lead.py (hunk 1 above)
Comment (toddcd) [reply clx_crm/models/crm_lead.py:140]: done

File: clx_crm/static/src/js/clx_crm.js (hunk 2 above)
Comment (toddcd) [reply clx_crm/static/src/js/clx_crm.js]: done

File: clx_ratio_invoice/static/src/js/clx_ratio_invoice.js (hunk 3 above)
Comment (toddcd) [reply clx_ratio_invoice/static/src/js/clx_ratio_invoice.js]: done

**Pull request merged**